add Go building clis introduction (#2361)
kamranahmedse/developer-roadmap: content/roadmaps/109-golang/content/102-go-building-clis/readme.md

> # Go building clis
> 
> ## CLI developers prefer Go for portability, performance, and ease of creation
> Command line interfaces (CLIs), unlike graphical user interfaces (GUIs), are text-only. Cloud and infrastructure applications are primarily CLI-based due to their easy automation and remote capabilities.
> 
> ## Key benefits
> ### Leverage fast compile times to build programs that start quickly and run on any system
> Developers of CLIs find Go to be ideal for designing their applications; from a single Windows or Mac laptop, developers can build a Go program for every one of the dozens of architectures and operating systems Go supports in a matter of seconds.
> 
> Go applications are built into a single self contained binary making installing Go applications trivial; specifically, programs written in Go run on any system without requiring any existing libraries, runtimes, or dependencies. And programs written in Go have an immediate startup time—similar to C or C++ but unobtainable with other programming languages.
> 
> <ResourceGroupTitle>Free Content</ResourceGroupTitle>
> <BadgeLink colorScheme='blue' badgeText='CLIs Overview' href='https://go.dev/solutions/clis'>Overview and use cases for CLIs</BadgeLink>

Reviewer: ```suggestion
# Building CLI Applications

Command line interfaces (CLIs), unlike graphical user interfaces (GUIs), are text-only. Cloud and infrastructure applications are primarily CLI-based due to their easy automation and remote capabilities.

Go applications are built into a single self contained binary making installing Go applications trivial; specifically, programs written in Go run on any system without requiring any existing libraries, runtimes, or dependencies. And programs written in Go have an immediate startup time—similar to C or C++ but unobtainable with other programming languages.

<ResourceGroupTitle>Free Content</ResourceGroupTitle>
<BadgeLink colorScheme='blue' badgeText='Read' href='https://go.dev/solutions/clis'>Command-line Interfaces (CLIs)</BadgeLink>
```
